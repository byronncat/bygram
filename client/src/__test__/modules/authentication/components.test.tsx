import { Brand, Form, FIELD, DEFAULT_VALUES } from '@authentication'
import renderer from 'react-test-renderer'

describe('components', () => {
  it('should render the Brand component correctly', () => {
    const brand = renderer.create(<Brand />)
    expect(brand).toMatchSnapshot()
  })

  it('should render the Form component correctly', () => {
    const submitHandler = jest.fn()

    const form = renderer.create(
      <Form
        fieldList={FIELD.LOGIN}
        defaultValues={DEFAULT_VALUES.LOGIN}
        submitHandler={submitHandler}
        submitPlaceholder="Login"
      />
    )
    expect(form).toMatchSnapshot()
  })
})
