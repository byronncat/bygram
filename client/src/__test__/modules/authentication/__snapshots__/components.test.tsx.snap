// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components should render the Brand component correctly 1`] = `
Array [
  <img
    alt="logo"
    className="d-block"
    height={60}
    src="logo.svg"
    width={60}
  />,
  <h1
    className="brand-name text-neon-glowing-1 text-uppercase fs-2 mt-2 my-3"
  >
    bygram
  </h1>,
]
`;

exports[`components should render the Form component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <input
      id="email"
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <label
      htmlFor="email"
    >
      Email
    </label>
    <p
      className=""
    />
  </div>
  <div>
    <input
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <span
      className="icon-eye"
      onClick={[Function]}
    >
      <i
        className="icon-eye"
      />
    </span>
    <p
      className=""
    />
  </div>
  <input
    className="submit-btn w-100 pt-2 my-2 btn"
    type="submit"
    value="Login"
  />
</form>
`;
